content/blog/2021-04-30-april-21-community-gems.md

```diff
@@ -0,0 +1,207 @@
+└───────────────┴──────────┴─────────┴────────────┴─────────────────┘
+```
+
+We are working on developing experiments to have features or documented patterns
+explicitly for grid search support, so definitely share any feedback to help
+drive the future direction of that!
+
+### [When importing/getting data from a repo, how do I provide credentials to the source repo remote storage without saving it into that Git repo?](https://discord.com/channels/485586884165107732/563406153334128681/830021022337073185)
+
+There's a bit of context behind this question that might give it more meaning.
+Here's the background information given by @EdAb in Discord:
+
+---
+
+I set up a private GitHub repo to be a data registry and I have set up a private
+Azure remote where I have pushed some datasets.
+
+I am now trying to read those datasets from another repository
+("my-project-repo"), using `dvc get` (e.g.
+`dvc get git@github.com:data-registry-repo.git path/data.csv`) but I get this
+error:
+
+```bash
+ERROR: failed to get 'path/data.csv' from 'git@github.com:data-registry-repo.git' - Authentication to Azure Blob Storage via default credentials (https://azuresdkdocs.blob.core.windows.net/$web/python/azure-identity/1.4.0/azure.identity.html#azure.identity.DefaultAzureCredential) failed.
+Learn more about configuration settings at <https://man.dvc.org/remote/modify>: unable to connect to account for Must provide either a connection_string or account_name with credentials!!
+```
+
+---
+
+Generally, there are two ways solve this issue:
+
+- [ENV vars](https://dvc.org/doc/user-guide/contributing/docs#env-variables)
+- Setup some options using the `--global` or `--system` flags to update the DVC
+  config
+
+If you remove the old remotes (with the same name) from your project repo and
+try to pull from the data registry using `AZURE_STORAGE_CONNECTION_STRING` and
+`AZURE_STORAGE_CONTAINER_NAME`, it should work!
+
+### [Is there any way to ensure that `dvc import` uses the cache from the config file?](https://discord.com/channels/485586884165107732/563406153334128681/827574712825413672)
+
+This is another great question where a little context might be useful.
+
+---
+
+I'm trying to import a dataset project into another DVC project, let's call it
+_dvcdata_.
+
+The config for _dvcdata_ is:
+
+```ini
+[core]
+    remote = awsremote
+[cache]
+    type = symlink
+    dir = /home/user/dvc_cache
+['remote "awsremote"']
+    url = s3://...
+```
+
+When I run `dvc import git@github.com:user/dvcdata.git my_data`, it starts to
+redownload it. I have double checked that I have pushed this config file to
+master.
+
+I have also checked that if I manually clone the repo and run
+`cd /tmp/dvcdata; dvc pull`, the files are symlinked.
+
+The command output says it's downloading to
+`../../../../../../../tmp/tmphlgqpv9zdvc-clone/<file name>` so the cache isn't
+being updated.
+
+---
+
+The repo you are importing into has its own cache directory. We recommend you
```

Comment:
we need to expand here - add examples (focus should be on explaining the solution, rather than on the context)

Reply:
@shcheklein I think I've got this one in a better state. I worked with a combination of @jorgeorpinel, @pmrowla, and your answers/feedback to come up with the answer. Let me know what you all think!